lib/src/solid/login.dart

```diff
@@ -230,172 +272,163 @@ class _SolidLoginState extends State<SolidLogin> {
 
     final webIdController = TextEditingController()..text = widget.webID;
 
-    // Define a common style for the text of the two buttons, GET POD and LOGIN.
-
-    const buttonTextStyle = TextStyle(
-      fontSize: 12.0,
-      // fontSize: MediaQuery.of(context).size.width * 0.03,
-      letterSpacing: 2.0,
-      fontWeight: FontWeight.bold,
-    );
-
     // The GET A POD button that when pressed will launch a browser to the
     // relevant link from where a user can register for a POD on the Solid
     // server. The default location is relative to the [webID], and is currently
     // a fixed path but needs to be obtained from the server meta data, as was
     // done in solid_auth through [getIssuer].
 
-    final registerButton = ElevatedButton(
-      // TODO 20231229 gjw NEED TO USE AN APPROACH TO GET THE RIGHT SOLID SERVER
-      // REGISTRATION URL WHICH HAS CHANGED OVER SERVERS. PERHAPS IT IS NEEDED
-      // TO BE OBTAINED FROM THE SERVER META DATA? CHECK WITH ANUSHKA. THE
-      // getIssuer() FROM solid-auth PERHAPS WITH lauchIssuerReg() COULD THEN BE
-      // USED AGAIN.
-
-      onPressed: () => launchUrl(
-          Uri.parse('${widget.webID}/.account/login/password/register/')),
-
-      child: Text(widget.registerText, style: buttonTextStyle),
+    final registerButton = PodButton(
```

Comment:
If the PodButton constructor accepts an instance of PodButtonStyle, it would save you three lines of code for each button?

Reply:
Hi @cdawei now we have 4 different data structure/ class called ContinueButtonStyle, LoginButtonStyle, RegisterButtonStyle and InfoButtonStyle because we need to keep 4 different default button styles(the button text names and tooltips are different). I am afraid we cannot make the PodButton constructor accepts 4 button styles?